syntax = "proto3";

package kvstore;

service KVStore {
	rpc Put (PutRequest) returns (PutReply);
	rpc Get (GetRequest) returns (GetReply);
	rpc Delete (DeleteRequest) returns (DeleteReply);	
}


message PutRequest {
	string key = 1;
	string value = 2;
}

message PutReply {
	bool success = 1;
}

message GetRequest {
	string key = 1;
}

message GetReply {
	bool found = 1;
	string value = 2;
}

message DeleteRequest {
	string key = 1;
}

message DeleteReply {
	bool success = 1;
}
